lib/srv/db/redis/client.go

```diff
@@ -101,30 +103,26 @@ type clusterClient struct {
 
 // newClient creates a new Redis client based on given ConnectionMode. If connection mode is not supported
 // an error is returned.
-func newClient(ctx context.Context, connectionOptions *connection.Options, tlsConfig *tls.Config, onConnect onClientConnectFunc) (redis.UniversalClient, error) {
+func newClient(ctx context.Context, connectionOptions *connection.Options, tlsConfig *tls.Config, credenialsProvider fetchCredentialsFunc) (redis.UniversalClient, error) {
 	connectionAddr := net.JoinHostPort(connectionOptions.Address, connectionOptions.Port)
 	// TODO(jakub): Investigate Redis Sentinel.
 	switch connectionOptions.Mode {
 	case connection.Standalone:
 		return redis.NewClient(&redis.Options{
-			Addr:      connectionAddr,
-			TLSConfig: tlsConfig,
-			OnConnect: onConnect,
-
-			// Auth should be done by the `OnConnect` callback here. So disable
-			// "automatic" auth by the client.
-			DisableAuthOnConnect: true,
```

Comment:
In the previous driver, we disabled auth on connect and performed auth during `OnConnect` callback.

We are switching to use `CredentialsProviderContext` now for auth and we set `Protocol` to v2 so the driver won't do RESP3 HELLO

Reply:
yes, the auth will be done using `HELLO 2 AUTH <user> <password>` instead of just AUTH. In my opinion, it's a win for newer servers. I haven't tested older version (<6.2) tho. The driver suppose to fallback to regular AUTH. We can test that during release testing.